src/loaders/tvg/tvgTvgLoader.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <fstream>
+#include <string.h>
+#include "tvgLoaderMgr.h"
+#include "tvgTvgLoader.h"
+#include "tvgTvgLoadParser.h"
+
+TvgLoader::~TvgLoader()
+{
+    close();
+}
+
+void TvgLoader::clearBuffer()
+{
+    this->size = 0;
+    if (this->buffer)
+    {
+        free(this->buffer);
+        this->buffer = nullptr;
+    }
+    this->pointer = nullptr;
+}
+
+bool TvgLoader::open(const string &path)
+{
+    ifstream f;
+    f.open(path, ifstream::in | ifstream::binary | ifstream::ate);
+
+    if (!f.is_open())
```

Comment:
please keep the coding convention. Also if it does nothing, you can return it at one line.
//LOG: Failed to open file
if (xx) return false;